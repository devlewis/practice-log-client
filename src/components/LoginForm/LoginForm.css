.LoginForm {
  margin: 0 5% 5% 5%;
  padding: 0 5% 5% 5%;
  background-color: aliceblue;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}

.LoginForm p {
  margin: 0;
  color: black;
}

.LoginForm div {
  margin-bottom: 1em;
}

.LoginForm button {
  width: fit-content;
  align-self: flex-end;
}

label {
  font-weight: 700;
  color: black;
  margin-bottom: 5px;
}

input,
label {
  width: 80%;
  font-size: 1em;
}

.LoginForm .red {
  color: red;
  font-size: 0.65em;
  margin: 0;
  padding: 0;
}

.daybox_DECO a {
  text-decoration: none;
}

@media screen and (min-width: 415px) {
  .LoginForm p,
  label,
  input {
    font-size: 0.9em;
  }
}
