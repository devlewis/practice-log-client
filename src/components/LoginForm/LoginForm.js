import React, { Component } from "react";
import AuthApiService from "../../services/auth-api-service";
import "./LoginForm.css";

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => { },
  };

  state = { error: null, user_name: "", password: "" };

  handleSubmitJwtAuth = (e) => {
    e.preventDefault();

    this.setState({ error: null });
    const { user_name, password } = this.state;

    AuthApiService.postLogin({
      user_name: user_name,
      password: password,
    })
      .then((res) => {
        this.setState({ user_name: "", password: "" });

        this.props.onLoginSuccess();
      })
      .catch((res) => {
        let error = "Could not reach server; please try again later!";
        if (res.error) {
          error = res.error;
        }
        this.setState({
          error: error,
        });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <div className="loginform_box">
        <form className="LoginForm" onSubmit={this.handleSubmitJwtAuth}>
          <div className="user_name">
            <label htmlFor="LoginForm__user_name">User name</label>
            <input
              required
              name="user_name"
              id="LoginForm__user_name"
              onChange={(e) => this.setState({ user_name: e.target.value })}
            ></input>
            <p>demo username: YoYoMa77!</p>
          </div>
          <div className="password">
            <label htmlFor="LoginForm__password">Password</label>
            <input
              required
              name="password"
              type="password"
              id="LoginForm__password"
              onChange={(e) => this.setState({ password: e.target.value })}
            ></input>
            <p>demo password: Cello123!</p>
          </div>
          <button type="submit">Login</button>
        </form>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
      </div>
    );
  }
}
