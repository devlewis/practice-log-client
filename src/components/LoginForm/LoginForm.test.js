import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoginForm from "./LoginForm";

describe(`LoginForm component`, () => {
  const props = {
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders form.LoginForm by default", () => {
    const wrapper = shallow(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the LoginForm given props", () => {
    const wrapper = shallow(<LoginForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
