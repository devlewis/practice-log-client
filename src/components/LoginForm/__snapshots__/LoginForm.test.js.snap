// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders form.LoginForm by default 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm__user_name"
    >
      User name
    </label>
    <input
      id="LoginForm__user_name"
      name="user_name"
      onChange={[Function]}
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      id="LoginForm__password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`LoginForm component renders the LoginForm given props 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm__user_name"
    >
      User name
    </label>
    <input
      id="LoginForm__user_name"
      name="user_name"
      onChange={[Function]}
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      id="LoginForm__password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;
